reflect/src/main/java/dagger/reflect/Scope.java

```diff
@@ -252,7 +252,8 @@ Builder addBindingElementsIntoSet(Key key, Binding elementsBinding) {
       if (key == null) throw new NullPointerException("key == null");
       if (elementsBinding == null) throw new NullPointerException("elementsBinding == null");
       if (Types.getRawType(key.type()) != Set.class) {
-        throw new IllegalArgumentException("key.type() must be Set");
+        throw new IllegalArgumentException(
+            "@BindsIntoSet must return Set. Found " + key.type() + ".");
```

Comment:
This still need reverted

Reply:
you mean keep it as `throw new IllegalArgumentException("key.type() must be Set");`?
isn't this always used in the case of `@BindsIntoSet`

Reply:
Yes. And yes, but `Scope` shouldn't know anything about how Dagger works.

Reply:
Done. Also updated the javadoc on this method.